### PR TITLE
Pass site or siteId to @netlify/config

### PR DESCRIPTION
**Pull request by RaeesBhatti**
**- Summary**

Fixes: https://github.com/netlify/cli/issues/959

**- Test plan**

**- Description for the changelog**

Use `siteId` or `site` flags as `siteId` as parameters to `resolveConfig`

**- A picture of a cute animal (not mandatory but encouraged)**
🐠

### CODE REVIEW

File: src/utils/command.js [hunk 1]
```diff
@@ -69,14 +69,14 @@ class BaseCommand extends Command {
...
-  async getConfig(cwd, state, token) {
+  async getConfig(cwd, state, token, argv) {
     try {
       return await resolveConfig({
         config: argv.config,
         cwd: cwd,
         context: argv.context,
         debug: argv.debug,
-        siteId: state.get('siteId'),
+        siteId: argv.siteId || typeof argv.site === 'string' && argv.site || state.get('siteId'),
```
Comment (ehmicky) [inline src/utils/command.js:79]: Do we have any CLI flag called `siteId` instead of `site`?

**Comment by ehmicky** [comment]
CI tests seem to be failing due to linting.

File: src/utils/command.js (hunk 1 above)
Comment (RaeesBhatti) [reply src/utils/command.js]: Yes, we have a `site` flag in `netlify deploy`

Comment (ehmicky) [reply src/utils/command.js]: We have a `site` CLI flag in other commands as well. But I was wondering about a `siteId` CLI flag?

Comment (RaeesBhatti) [reply src/utils/command.js]: The other `site` flag is a boolean. That's why I'm checking for `typeof`. We're checking for both `siteId` and a string `site` flag now.

Comment (ehmicky) [reply src/utils/command.js]: I am wondering if `argv.siteId ||` could be removed since we don't have any `siteId` CLI flag at the moment?

**Pull request merged**